### PR TITLE
chore: rm'd reflective buildModel & replaced project by content as disposable (223#)

### LINKED ISSUE #223: tool windows: remove reflective buildModel and use legal disposable
related to https://github.com/redhat-developer/intellij-kubernetes/issues/498

* `KnBaseWindowTool.buildModel` was needed back then when our least supported version was 2018. We're now requiring at least 2021.1
see here: https://github.com/redhat-developer/intellij-common/blob/main/src/main/java/com/redhat/devtools/intellij/common/tree/StructureTreeModelFactory.java#L39
```java
try {
	Constructor<StructureTreeModel> constructor = StructureTreeModel.class.getConstructor(AbstractTreeStructure.class);
	return constructor.newInstance(structure);
} catch (NoSuchMethodException e) {
	// IC 2019.3+
	Constructor<StructureTreeModel> constructor = StructureTreeModel.class.getConstructor(AbstractTreeStructure.class, Disposable.class);
	return constructor.newInstance(structure, project);
}
```
* `StructureTreeModel` and `AsyncTreeModel` are both provided with the `project` as disposable. The [SDK documentation](https://plugins.jetbrains.com/docs/intellij/disposers.html#choosing-a-disposable-parent) states the `Project` should not be used as disposable because this causes memory leaks:

	> Even though Application and Project implement Disposable, they must NEVER be used as parent disposables in plugin code. Disposables registered using those objects as parents will not be disposed when the plugin is unloaded, leading to memory leaks.

### PR DESCRIPTION
**Pull request by adietish**
fixes #223

### CODE REVIEW

**Comment by adietish** [comment]
@lstocchi can you please review this change? thanks.

File: src/main/java/com/redhat/devtools/intellij/knative/ui/toolwindow/FunctionsWindowToolFactory.java [hunk 1]
```diff
@@ -26,11 +28,12 @@ public class FunctionsWindowToolFactory extends KnBaseWindowTool<KnTreeStructure
     @Override
     public void createToolWindowContent(@NotNull Project project, @NotNull ToolWindow toolWindow) {
         setTitleAndIcon(toolWindow, "Functions");
+        SimpleToolWindowPanel panel = new SimpleToolWindowPanel(true, true);
+        Content content = createContent(toolWindow, panel, FUNCTIONS_TOOLBAR_ACTION_GROUP_ID);
 
         KnFunctionsTreeStructure structure = new KnFunctionsTreeStructure(project);
-        Tree tree = createTree(project, structure, true);
+        Tree tree = createTree(content, structure, true, panel);
```
[the remark this answers is not in the document]
Comment (lstocchi) [reply src/main/java/com/redhat/devtools/intellij/knative/ui/toolwindow/FunctionsWindowToolFactory.java:35]: If you're busy with other stuff, let me know. I'll edit this PR myself. Thanks for the help!!

[the remark this answers is not in the document]
Comment (adietish) [reply src/main/java/com/redhat/devtools/intellij/knative/ui/toolwindow/FunctionsWindowToolFactory.java:35]: @lstocchi: I added you to my fork. You can add your changes on top of my commit onto my repo?

[the remark this answers is not in the document]
Comment (lstocchi) [reply src/main/java/com/redhat/devtools/intellij/knative/ui/toolwindow/FunctionsWindowToolFactory.java:35]: Sure :+1:

**Comment by adietish** [comment]
@lstocchi LGTM

**Comment by lstocchi** [comment]
Thanks @adietish :pray:

**Pull request merged**